src/classes/OPP_StageMappingUtil.cls

```diff
@@ -91,10 +91,53 @@ public with sharing class OPP_StageMappingUtil {
     * necessary.
     ********************************************************************************************************/
     @testVisible private static void initializeStates() {
-        // These are temporary values to be used only during interim for internal development on Abacus project.
-        // This will be updated in March 2019 timeframe with revised code to retrieve from Custom Metadata instead
-        COMMITTED_STATE_SET = new Set<String> {'closed won'};
-        FINALIZED_STATE_SET = new Set<String> {'finalized'};
+
+        // If either the committed state set or the finalized state set is null, query for everything and 
+        // reinitialize everything. 
+        if (COMMITTED_STATE_SET == null || FINALIZED_STATE_SET == null) {
+            Map<String, Set<String>> oppStateStages = new Map<String, Set<String>>();
+            List<Opportunity_Stage_To_State_Mapping__mdt> stageToStateRecords = getOpportunityStageToStateMappings();
+            
+            for (Opportunity_Stage_To_State_Mapping__mdt mappingRecord : stageToStateRecords) {
+                if (mappingRecord.Opportunity_State__c != null) {
+
+                    String oppState = mappingRecord.Opportunity_State__c.trim().toUpperCase();
+                    String oppStage = mappingRecord.Opportunity_Stage__c.trim().toLowerCase();
+                    Set<String> stages = oppStateStages.containsKey(oppState) ? oppStateStages.get(oppState) : new Set<String>();
+                    stages.add(oppStage);
+                    oppStateStages.put(oppState, stages);
+                }
+            }
+            
+            COMMITTED_STATE_SET = oppStateStages.containsKey(COMMITTED_STATE_VAL) ? oppStateStages.get(COMMITTED_STATE_VAL) : new Set<String>();
+            FINALIZED_STATE_SET = oppStateStages.containsKey(FINALIZED_STATE_VAL) ? oppStateStages.get(FINALIZED_STATE_VAL) : new Set<String>();
+        }
     }
 
+    /*******************************************************************************************************
+    * @description Returns a list of Opportunity stage to state mappings. If it is a test running,
+    * return an artificially created list. Otherwise, get it from the database
+    ********************************************************************************************************/
+    private static List<Opportunity_Stage_To_State_Mapping__mdt> getOpportunityStageToStateMappings() {
+        List<Opportunity_Stage_To_State_Mapping__mdt> metadataRecordList = new List<Opportunity_Stage_To_State_Mapping__mdt>();
+
+        if (Test.isRunningTest()) {
+            
+            // If it's a test running, return artificial records. This ensures consistency in the testing. 
+            Opportunity_Stage_To_State_Mapping__mdt finalizedRecord = new Opportunity_Stage_To_State_Mapping__mdt();
+            finalizedRecord.Opportunity_Stage__c = 'Closed Won';
```

Comment:
Can you make these static final Strings at the top, and call them TEST_COMMITTED_STAGE and TEST_FINALIZED_STAGE?